test/benchmarks/driver_bench/src/runner.mts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-console */
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import process from 'node:process';
+
+import { metrics, snakeToCamel } from './driver.mjs';
+
+const [, , benchmarkFile] = process.argv;
+
+type BenchmarkModule = {
+  taskSize: number;
+  before?: () => Promise<void>;
+  beforeEach?: () => Promise<void>;
+  run: () => Promise<void>;
+  afterEach?: () => Promise<void>;
+  after?: () => Promise<void>;
+};
+
+const benchmarkName = snakeToCamel(path.basename(benchmarkFile, '.mjs'));
+const benchmark: BenchmarkModule = await import(`./${benchmarkFile}`);
+
+if (typeof benchmark.taskSize !== 'number') throw new Error('missing taskSize');
+if (typeof benchmark.run !== 'function') throw new Error('missing run');
+
+/** CRITICAL SECTION: time task took in seconds */
+async function timeTask() {
+  const start = performance.now();
+  await benchmark.run();
+  const end = performance.now();
+  return (end - start) / 1000;
+}
+
+/** 1 min in seconds */
+const ONE_MIN = 1 * 60;
+/** 5 min in seconds */
+const FIVE_MIN = 5 * 60;
+/** Don't run more than 100 iterations */
+const MAX_COUNT = 100;
+
+await benchmark.before?.();
+
+// for 1/10th the max iterations
+const warmupIterations = (MAX_COUNT / 10) | 0;
+
+// Warm Up.
+for (let i = 0; i < warmupIterations; i++) {
+  await benchmark.beforeEach?.();
+  await timeTask();
+  await benchmark.afterEach?.();
+}
+
+// Allocate an obscene amount of space
+const data = new Float64Array(10_000_000);
```

Comment:
I take it this is because we don't necessarily know how many times we're going to run the test?

Reply:
even returnDocument (which we're going to slow down) runs for significantly less than this amount (at least on my machine) so I feel confident we'll never use this much, and we have the error checking to tell us if we do.